code/espurna/garland/anim_spread.cpp

```diff
@@ -0,0 +1,46 @@
+#if GARLAND_SUPPORT
+
+#include "anims.h"
+#include "color.h"
+#include "palette.h"
+#include "scene.h"
+
+#define SPREAD_MAX_WIDTH 20
+
+AnimSpread::AnimSpread() : Scene::Anim("Spread") {
+}
+
+void AnimSpread::SetupImpl() {
+    inc = random(2,8);
+    memset(seq, 0, LEDS);
+}
+
+void AnimSpread::Run() {
+    memset(_leds, 0, 3*LEDS);
```

Comment:
I'd try to avoid memset when possible for objects. It is a valid C-pattern, but not really in C++ where `struct` aka `class` can have a constructor attached (and note there's no real difference between the two, only in the requirement for explicit `public:` or `private:` at the start of the body).